bc_obps/common/service/email/email_service.py

```diff
@@ -1,27 +1,37 @@
 from datetime import datetime, timedelta
-from typing import Optional
+from typing import List, Optional
 from uuid import UUID
-import logging
-
-import requests
+from common.enums import AdminAccessRequestStates
+import logging, requests
+from common.models import EmailNotification, EmailNotificationTemplate
 from django.conf import settings
 
 logger = logging.getLogger(__name__)
 
+SENDER_EMAIL = 'no-reply.cas@gov.bc.ca'
+
 
-class EmailService:
+class EmailService(object):
     """
     EmailService uses Common Hosted Email Service (CHES) API to enqueue emails for delivery. Uses BC Government-hosted SMTP server to send emails.
     """
 
-    def __init__(self):
-        self.api_url: str = settings.CHES_API_URL
-        self.client_id: str = settings.CHES_CLIENT_ID
-        self.client_secret: str = settings.CHES_CLIENT_SECRET
-        self.token_endpoint: str = settings.CHES_TOKEN_ENDPOINT
-        self.token: Optional[str] = None
-        self.token_expiry: datetime = datetime.now()
-        logger.info(f'Initializing EmailService for clientID {self.client_id} to connect to {self.api_url}')
+    _instance = None
```

Comment:
this is extremely nitpicky (but I'm in a nitty headspace because I spent all afternoon in interviews) - I'd suggest adding an extra line of documentation/comment here explaining how to use the EmailService class in other files. I think that would be helpful to future devs.

I think something as simple as this would suffice:
"Use `email_service = EmailService()` to access the EmailService singleton object in other .py files"

Reply:
This is a very valid point, thank you 🙏